File: src/javatest/huff/Node.java
package javatest.huff;

import java.util.*;

/**
 * Created by wq on 16-4-23.
 */
public class Node<E> implements Comparable<Node<E>> {
    E data;
    Node<E> leftNode;
    Node<E> rightNode;
    double weight;

    public Node(E data,double weight){
        this.data = data;
        this.weight = weight;
    }

    public void setWeight(double weight){
        this.weight = weight;
    }

    public double getWeight(){
        return weight;
    }

    public int compare(Node x, Node y) {
        return (x.getWeight() < y.getWeight()) ? -1 : ((x.getWeight() == y.getWeight()) ? 0 : 1);
    }

    public int compareTo(Node node){
        return compare(this,node);
    }

}

File: src/scalaMod/huff/Node.scala
package scalaMod.huff

import scalaMod.huff._

/**
 * Created by wq on 16-4-25.
 */
class Node(val data:String,val weight:Double) extends Comparable[Node]{
  var leftNode:Node=null
  var rightNode:Node=null

  def compare(x:Node,y:Node):Int ={
    val in:Int = if(x.weight<y.weight) -1 else if(x.weight == y.weight) 0 else 1
    in
  }

  override def compareTo(node:Node):Int = {
    val i = compare(this,node)
    i
  }
}